### PR TITLE
Added Authentication to no-framework-rooms

### PR DESCRIPTION
**Pull request by Ishan27g**
# Added Authentication to no-framework-rooms

#### Can merge this branch into `no-framework-rooms` before merging `no-framework-rooms` into `main`

- Added signup/login, same as previous PR
- Not sure how to compile `client/auth.ts` so I added it to `package.json`

#### Changes:

- If you go to` /, `you will be served `auth.html` where you can login or signup
- Successful login will redirect to `join.html` with a token, presence of token is assumed to be a valid login, no verification of JWT
`join.html` has two columns. 
- Clicking 'Create' generates a random room id, (_would connect to peerServer for its things -> yet to be integrated as I didn't quite understand the /:room endpoint_) and a custom room can be joined via Join column.
- Entering a room id and joining it forwards it to `/:roomId`

#### Yet to be tested:

- Actual video call, cannot see any video on screen
- Integration with creating the room

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "start": "NODE_ENV=production node server/server.js",
-    "dev:server": "ts-node-dev --respawn --pretty --project server/tsconfig.json server/server.ts",
+    "dev:server": "tsc client/scripts/auth.ts && ts-node-dev --respawn --pretty --project server/tsconfig.json server/server.ts",
```
Comment (dalyIsaac) [inline package.json]: I'm not sure this change is required, `npm run dev:client` should compile `auth.ts` too.

File: server/server.ts [hunk 2]
```diff
@@ -21,8 +21,16 @@ const peerServer = PeerWrapper.init(httpServer, rooms);
 app.use("/peerjs", peerServer.peerExpressApp);
 app.use(express.json());
 app.use(express.static(CLIENT_PATH));
+app.use(express.static(path.join(__dirname,CLIENT_PATH+'/style')))
```
Comment (dalyIsaac) [inline server/server.ts]:
```suggestion
app.use(express.static(path.join(__dirname, CLIENT_PATH + "/style")));
```

formatting

File: client/auth.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                </div>
+            </div>
+        </div>
+    </div>
+    <h3 id="ActionStatus"></h3>
+</body>
+<script src="scripts/auth.js"></script>
+<script>
+    // Zoomutex server url
+    const zoomMutexJoin = "http://127.0.0.1:7000/join"
```
Comment (dalyIsaac) [inline client/auth.html]: We'll have to change this for deploying to heroku, right?

**Review by dalyIsaac — CHANGES_REQUESTED** [review]
Clicking the `Signup` button causes the page to reload, so the `serverRspStatus` is never updated. 

As a result, `signup` is cancelled:

image

This prevents a user from signing up.

File: package.json (hunk 1 above)
Comment (Ishan27g) [reply package.json]: I run the only the server, I thought the server is serving the client. 
I realise now that I'm supposed to run both the dev: commands in a separate terminal

File: client/auth.html (hunk 3 above)
Comment (Ishan27g) [reply client/auth.html]: Yep, i'll add that in.

**Comment by Ishan27g** [comment]
I fixed the broken signup stuff. I've deployed `no-framework-rooms-auth` to  Heroku, looks good to me

**Comment by Ishan27g** [comment]
Not sure why login doesn't work on the first attempt. Everything else looks fine `https://zoomutex.herokuapp.com/`

**Comment by Ishan27g** [comment]
seems to be working now

**Comment by Ishan27g** [comment]
Passing token as a url param doesnt work when deployed to Heroku, works perfectly fine locally.  Heroku url shows '?token=undefined'.

**Comment by dalyIsaac** [comment]
@Ishan27g I've had another look, and I still have intermittent issues with signing in.
image

I'm not sure what the point of retaining the authentication is when it can be ignored - we could just drop the authentication functionality since the rooms functionality is working.

Do you have any thoughts?

**Comment by Ishan27g** [comment]
> @Ishan27g I've had another look, and I still have intermittent issues with signing in.
> image
> 
> I'm not sure what the point of retaining the authentication is when it can be ignored - we could just drop the authentication functionality since the rooms functionality is working.
> 
> Do you have any thoughts?

Yep sounds appropriate. I'll make the changes and remove the login/signup part

**Comment by Ishan27g** [comment]
Removed authentication and deployed it

**Pull request merged**